### PR TITLE
docs: Add AUHTORS.md into TiDB

### PR DESCRIPTION
**Pull request by gaocegege**
## Motivation

GitHub only shows 100 contributors in [pingcap/tidb/graphs/contributors](https://github.com/pingcap/tidb/graphs/contributors), but every contributor is unique in the community, so I think it is time to import AUTHORS into TiDB.

## Generation

```bash
cd ${GOPATH}/src/github.com/pingcap/tidb
go get github.com/gaocegege/maintainer
maintainer contributor
```

## Order

Now the file is generated by `git log --format='%aN <%aE>'`. Is it necessary to reorder the output alphabetically?

### CODE REVIEW

**Comment by ngaut** [comment]
I think we should keep different role of authors. How about add three sections ? Maintainer Committer and contributor.

**Comment by gaocegege** [comment]
@ngaut Then we need to update the list manually, IMO.

**Comment by ngaut** [comment]
You are right, we can keep committers and maintainers in another file.
Also we should keep the authors order by commit count, let's keep consistent with github.

**Comment by gaocegege** [comment]
OK, I will file a new commit soon.

**Comment by hanfei1991** [comment]
hanfei1991, hanfei, Fei Han, Han Fei, hanfei19910905 is the same boy.

**Comment by gaocegege** [comment]
The result is a little different from github, it seems that GitHub doesn't treat merge commit as a commit.

See https://pastebin.com/7AFCGrQq

**Comment by gaocegege** [comment]
We need to use email as ID, not name + email, I will update the doc soon, again.

**Comment by gaocegege** [comment]
PTAL https://github.com/gaocegege/tidb/blob/f2ea78945e40640008673e352fa530462adf6af2/AUTHORS.md

Abandon the git way, use GitHub API to get all contributors.

File: AUTHORS.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# Authors
+
+This list is sorted by the number of commits per contributor in descending order.
```
Comment (ngaut) [inline AUTHORS.md]: s/This list is sorted by the number of commits per contributor in descending order/The list is auto-generated by xxx scripts, please do not edit. The list is sorted by the number of commits per contributor in descending order/

File: AUTHORS.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+[@weekface](https://github.com/weekface)
+
+[@zs634134578](https://github.com/zs634134578)
+
+[@qiukeren](https://github.com/qiukeren)
+
+[@hawkingrei](https://github.com/hawkingrei)
+
+###### Auto generated by [gaocegege/maintainer](https://github.com/gaocegege/maintainer) on 2017-05-22
```
Comment (ngaut) [inline AUTHORS.md]: Please remove the two lines.

Comment (gaocegege) [inline AUTHORS.md]: This is auto-generated, if remove, remember to remove the text every time we update the doc.

**Comment by gaocegege** [comment]
PTAL

**Comment by ngaut** [comment]
PTAL @queenypingcap

**Comment by shenli** [comment]
LGTM

**Comment by zhexuany** [comment]
@shenli @ngaut Any particular reason for the extension `.md`. Many open source project does not have such extension. Instead, `Authors`  will be enough. for example https://github.com/bazelbuild/bazel/blob/master/AUTHORS

**Comment by gaocegege** [comment]
I think markdown has better human readability.

**Comment by shenli** [comment]
@gaocegege I am agree with you.

**Comment by zhexuany** [comment]
I did not say markdown is bad. Github can still render md file even without
the extension. I was suggesting replace Authors.md with Authors.
​

**Comment by gaocegege** [comment]
I know, but markdown supports URL link, which we could use to contributors' GitHub profile page.

And IMO, people use plain text, not markdown because AUTHORS is an old thing, there is no markdown at that time, so open source developers still use plain text today. And you know README is in plain text before markdown get popular, too, now we are all use markdown, not plain text to write it.

If we use plain text, instead, then we couldn't click the contributors' name or email to trigger a GitHub redirect or a <mailto:> link.

**Comment by zhexuany** [comment]
Wow. That is cool. I did not know that. lol. point taken.

**Comment by gaocegege** [comment]
I have added all contributors in one `<ul>` list, PTAL

This will be the final version, IMO.

**Comment by gaocegege** [comment]
@zhexuany :clinking_glasses: 

There are some new repos which uses markdown to write AUTHORS now :smile: We are "第一个吃螃蟹的人"

**Comment by ngaut** [comment]
LGTM

**Pull request merged**